plugins/loop_registry.go

```diff
@@ -74,6 +75,8 @@ func (m *LoopRegistry) Register(id string) (*RegisteredLoop, error) {
 		envCfg.TelemetryCACertFile = m.cfgTelemetry.CACertFile()
 		envCfg.TelemetryAttributes = m.cfgTelemetry.ResourceAttributes()
 		envCfg.TelemetryTraceSampleRatio = m.cfgTelemetry.TraceSampleRatio()
+		envCfg.TelemetryAuthHeaders = beholder.GetAuthenticator().GetHeaders()
+		envCfg.TelemetryAuthPubKeyHex = fmt.Sprintf("%x", beholder.GetAuthenticator().GetPubKey())
```

Comment:
I think these should be directly injected to the constructor to avoid any future confusion about ordering.

Reply:
Sorry could you please clarify? I read this as suggesting to pass the value from `beholder.GetAuthenticator().GetHeaders()`  when calling `NewLoopRegistry`, but I think I misunderstood given [your reply](https://github.com/smartcontractkit/chainlink-common/pull/877#discussion_r1828416340) on the chainlink-common PR

Reply:
Can we pass around exclusively things that we construct directly, instead of communicating through globals?